### PR TITLE
feat: migrate google-cloud-datastore/java into java-datastore

### PR DESCRIPTION
**Pull request by kolea2**
source: https://github.com/googleapis/google-cloud-datastore/tree/master/java/datastore

### CODE REVIEW

**Comment by kolea2** [comment]
@BenWhitehead I didn't include the parent pom in https://github.com/googleapis/google-cloud-datastore/tree/master/java and instead switched the parent to this repo's pom. Let me know if you have any concerns.

**Review by BenWhitehead — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -172,7 +172,7 @@
       <dependency>
         <groupId>com.google.cloud.datastore</groupId>
         <artifactId>datastore-v1-proto-client</artifactId>
-        <version>1.6.3</version>
+        <version>1.6.4-SNAPSHOT</version><!-- {x-version-update:datastore-v1-proto-client:current} -->
         <exclusions>
```
Comment (BenWhitehead) [inline pom.xml:176]: You may be able to remove this exclusion now that the protos are from this build instead of from maven central.

**Comment by BenWhitehead** [comment]
LGTM, changing parent poms should be fine. Most of the dependencies for the datastore-v1 client were bom managed anyway at this point.

One small comment where the pom may be able to be cleaned up, but not a blocker.

**Review by chingor13 — APPROVED** [review]
Ensure to regular merge or rebase merge if you want to keep individual commit histories.

**Pull request merged**